client/src/legacy/LeftAndMain.js

```diff
@@ -1086,6 +1086,14 @@ $.entwine('ss', function($) {
     }
   });
 
+  $('.cms button.cms-panel-link').entwine({
+    onclick: function(e) {
+      if (!$('.cms-container').checkCanNavigate()) {
+        e.preventDefault();
+      }
+    }
+  });
+
```

Comment:
I'm quite certain you don't need these click handlers. Just remove `cms-panel-link` from the buttons. They're form actions, so they already have ajax submit handlers attached to them. I've done a really quick test, and it works fine if you remove that class and remove these changes.

Reply:
Gotcha. I still don't see why it needs to be `cms-panel-link`, though. That class is used for a very specific purpose, and adding it to your form actions adds complexity and potential breakages. What is `cms-panel-link` doing for you here?

Reply:
It is necessary to distinguish from actions that don't need the discard confirmation (such as the save action).

I have updated this with a more relevant class name.